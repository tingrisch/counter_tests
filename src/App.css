.App {
  margin: 50px auto;
  width: 300px;
  text-align: center;
}

.App * {
  font: 26pt Verdana;
  padding: 10px;
  display: inline;
}
