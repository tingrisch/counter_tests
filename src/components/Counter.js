import React from "react";
import styled from "styled-components";

const CounterDiv = styled.div`
  font-family: Arial;
  font-size: 22pt;
  color: ${props => props.count < 0 ? 'red' : 'green'};
`;

const Counter = ({ count }) =>
  <CounterDiv count={count}>
    {count}
  </CounterDiv>;

export default Counter;
