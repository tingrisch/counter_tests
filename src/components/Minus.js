import React from "react";
import Button from "../styled/Button"

const Minus = ({ decrement }) => <Button onClick={decrement}>-</Button>;

export default Minus;
