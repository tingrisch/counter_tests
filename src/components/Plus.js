import React from "react";
import Button from "../styled/Button"

const Plus = ({ increment }) => <buttons primary onClick={increment}>+</buttons>;

export default Plus;
