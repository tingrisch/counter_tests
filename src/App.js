import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { increment, decrement } from "./actions/counterActions";

import "./App.css";
import Counter from "./components/Counter";
import Plus from "./components/Plus";
import Minus from "./components/Minus";

import styled from "styled-components"

const Wrapper = styled.div`
  border: 2px solid blue;
  background-color: palevioletred;
  padding: 20px;
`;

const App = ({ count, decrement, increment }) =>
  <Wrapper className="App">
    <Plus decrement={decrement} />
    <Counter count={count} />
    <Minus increment={increment} />
  </Wrapper>;

const mapStateToProps = state => ({
  count: state.count
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ decrement, increment }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);

// For Testing, export unconnected Component
export { App };
