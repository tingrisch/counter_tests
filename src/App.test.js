import React from "react";
import { mount } from "enzyme";

import { App } from "./App";
import Plus from "./components/Plus";
import Minus from "./components/Minus"
import Counter from "./components/Counter";

import renderer from "react-test-renderer";

describe("App Component", () => {
  let wrapper;
  // this resets our component to a clean state before each test
  // ensuring that tests are not interefering with each other
  beforeEach(() => {
    wrapper = mount(<App count={0} />);
  });

  it("handles a click on Plus", () => {
    const spy = jest.fn(); //gibt einfach eine Funktion zurÃ¼ck, die benutzt werden kann
    wrapper.setProps({
      increment: spy
    });
    //console.log(wrapper.debug()); // anschauen, was die Component gerade gerendert hat
    wrapper.find(Plus).simulate("click");
    expect(spy).toBeCalled();
  });

  it("handles a click on Minus", () => {

    const spy = jest.fn();
    wrapper.setProps({
      decrement: spy
    });

    wrapper.find(Minus).simulate("click");
    expect(spy).toBeCalled();
  });

  it("renders all sub components correctly", () => {
    // render tree to JSON
    // expect snapshot to match
    const tree = renderer.create(<App count={2} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
