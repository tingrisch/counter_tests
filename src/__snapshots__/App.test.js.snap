// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders all sub components correctly 1`] = `
<div
  className="App sc-htpNat jxIsVz"
>
  <button
    className="sc-bwzfXH jqnzSp"
    onClick={undefined}
  >
    +
  </button>
  <div
    className="sc-bdVaJa jeITAu"
  >
    2
  </div>
  <button
    className="sc-bwzfXH gEpyaL"
    onClick={undefined}
  >
    -
  </button>
</div>
`;
